SQS IntegrationSink queueNameOrArn missing

**Describe the bug**
The SQS IntegrationSink can not be successfully created. Once running, I am presented this error within the associated pod:
`
...
Caused by: java.lang.IllegalArgumentException: Route template aws-sqs-sink the following mandatory parameters must be provided: queueNameOrArn
	at org.apache.camel.impl.DefaultModel.addRouteFromTemplate(DefaultModel.java:478)
	at org.apache.camel.impl.DefaultModel.addRouteFromTemplate(DefaultModel.java:416)
	at org.apache.camel.impl.DefaultCamelContext.addRouteFromTemplate(DefaultCamelContext.java:362)
	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.createRouteForEndpoint(KameletComponent.java:457)
	... 28 more`





 **Expected behavior**
The IntegrationSink is created and runs correctly with no errors.

 **To Reproduce**

```apiVersion: sinks.knative.dev/v1alpha1
kind: IntegrationSink
metadata:
  name: sqsIntegrationSinkTest
  namespace: test
spec:
  aws:
    auth:
      secret:
        ref:
          name: sqsIntegrationSinkTest
    sqs:
      arn: "arn:aws:sqs:us-east-2:************:sqsIntegrationSinkTest
      region: us-east-2
```

 **Knative release version**
1.17.0

 **Additional context**
This is the first time attempting to use this custom resource type. I noticed on the describe call against the associated pod this set of registered environment variables:

```    Environment:
      CAMEL_KAMELET_AWS_SQS_SINK_REGION:              us-east-2
      CAMEL_KAMELET_AWS_SQS_SINK_OVERRIDEENDPOINT:    false
      CAMEL_KAMELET_AWS_SQS_SOURCE_QUEUENAMEORARN:    arn:aws:sqs:us-east-2:************:sqsIntegrationSinkTest
      CAMEL_KAMELET_AWS_SQS_SINK_DELETEAFTERREAD:     true
      CAMEL_KAMELET_AWS_SQS_SINK_AUTOCREATEQUEUE:     false
      CAMEL_KAMELET_AWS_SQS_SOURCE_AMAZONAWSHOST:     amazonaws.com
      CAMEL_KAMELET_AWS_SQS_SINK_PROTOCOL:            https
      CAMEL_KAMELET_AWS_SQS_SINK_GREEDY:              false
      CAMEL_KAMELET_AWS_SQS_SINK_DELAY:               500
      CAMEL_KAMELET_AWS_SQS_SINK_MAXMESSAGESPERPOLL:  1
      CAMEL_KAMELET_AWS_SQS_SINK_WAITTIMESECONDS:     0
      CAMEL_KAMELET_AWS_SQS_SINK_VISIBILITYTIMEOUT:   0
      CAMEL_KAMELET_AWS_SQS_SINK_ACCESSKEY:           <set to the key 'aws.accessKey' in secret '*******'>  Optional: false
      CAMEL_KAMELET_AWS_SQS_SINK_SECRETKEY:           <set to the key 'aws.secretKey' in secret '*******'>  Optional: false
```


I believe that the environment variable `CAMEL_KAMELET_AWS_SQS_SOURCE_QUEUENAMEORARN"`should be `CAMEL_KAMELET_AWS_SQS_SINK_QUEUENAMEORARN` in order for this code

```	if sink.Spec.Log != nil {
		envVars = append(envVars, integration.GenerateEnvVarsFromStruct("CAMEL_KAMELET_LOG_SINK", *sink.Spec.Log)...)
		return envVars
	}
```
 [here](https://github.com/knative/eventing/blob/3153b4735d1c5dbfc9e3c0bfa98e655219fe0708/pkg/reconciler/integration/sink/resources/container_image.go#L166) to properly read the environment variables.

--- a/pkg/apis/common/integration/v1alpha1/aws.go
+++ b/pkg/apis/common/integration/v1alpha1/aws.go
@@ -33,33 +33,33 @@ type AWSCommon struct {
 
 type AWSS3 struct {
 	AWSCommon               `json:",inline"` // Embeds AWSCommon to inherit its fields in JSON
-	Arn                     string           `json:"arn,omitempty" camel:"CAMEL_KAMELET_AWS_S3_SOURCE_BUCKETNAMEORARN"` // S3 ARN
-	DeleteAfterRead         bool             `json:"deleteAfterRead" default:"true"`                                    // Auto-delete objects after reading
-	MoveAfterRead           bool             `json:"moveAfterRead" default:"false"`                                     // Move objects after reading
-	DestinationBucket       string           `json:"destinationBucket,omitempty"`                                       // Destination bucket for moved objects
-	DestinationBucketPrefix string           `json:"destinationBucketPrefix,omitempty"`                                 // Prefix for moved objects
-	DestinationBucketSuffix string           `json:"destinationBucketSuffix,omitempty"`                                 // Suffix for moved objects
-	AutoCreateBucket        bool             `json:"autoCreateBucket" default:"false"`                                  // Auto-create S3 bucket
-	Prefix                  string           `json:"prefix,omitempty"`                                                  // S3 bucket prefix for search
-	IgnoreBody              bool             `json:"ignoreBody" default:"false"`                                        // Ignore object body
-	ForcePathStyle          bool             `json:"forcePathStyle" default:"false"`                                    // Force path style for bucket access
-	Delay                   int              `json:"delay" default:"500"`                                               // Delay between polls in milliseconds
-	MaxMessagesPerPoll      int              `json:"maxMessagesPerPoll" default:"10"`                                   // Max messages to poll per request
+	Arn                     string           `json:"arn,omitempty" camel:"BUCKETNAMEORARN"` // S3 ARN
+	DeleteAfterRead         bool             `json:"deleteAfterRead" default:"true"`        // Auto-delete objects after reading
+	MoveAfterRead           bool             `json:"moveAfterRead" default:"false"`         // Move objects after reading
+	DestinationBucket       string           `json:"destinationBucket,omitempty"`           // Destination bucket for moved objects
+	DestinationBucketPrefix string           `json:"destinationBucketPrefix,omitempty"`     // Prefix for moved objects
+	DestinationBucketSuffix string           `json:"destinationBucketSuffix,omitempty"`     // Suffix for moved objects
+	AutoCreateBucket        bool             `json:"autoCreateBucket" default:"false"`      // Auto-create S3 bucket
+	Prefix                  string           `json:"prefix,omitempty"`                      // S3 bucket prefix for search
+	IgnoreBody              bool             `json:"ignoreBody" default:"false"`            // Ignore object body
+	ForcePathStyle          bool             `json:"forcePathStyle" default:"false"`        // Force path style for bucket access
+	Delay                   int              `json:"delay" default:"500"`                   // Delay between polls in milliseconds
+	MaxMessagesPerPoll      int              `json:"maxMessagesPerPoll" default:"10"`       // Max messages to poll per request
 }
 
 type AWSSQS struct {
 	AWSCommon          `json:",inline"` // Embeds AWSCommon to inherit its fields in JSON
-	Arn                string           `json:"arn,omitempty" camel:"CAMEL_KAMELET_AWS_SQS_SOURCE_QUEUENAMEORARN"`               // SQS ARN
-	DeleteAfterRead    bool             `json:"deleteAfterRead" default:"true"`                                                  // Auto-delete messages after reading
-	AutoCreateQueue    bool             `json:"autoCreateQueue" default:"false"`                                                 // Auto-create SQS queue
-	Host               string           `json:"host" camel:"CAMEL_KAMELET_AWS_SQS_SOURCE_AMAZONAWSHOST" default:"amazonaws.com"` // AWS host
-	Protocol           string           `json:"protocol" default:"https"`                                                        // Communication protocol (http/https)
-	QueueURL           string           `json:"queueURL,omitempty"`                                                              // Full SQS queue URL
-	Greedy             bool             `json:"greedy" default:"false"`                                                          // Greedy scheduler
-	Delay              int              `json:"delay" default:"500"`                                                             // Delay between polls in milliseconds
-	MaxMessagesPerPoll int              `json:"maxMessagesPerPoll" default:"1"`                                                  // Max messages to return (1-10)
-	WaitTimeSeconds    int              `json:"waitTimeSeconds,omitempty"`                                                       // Wait time for messages
-	VisibilityTimeout  int              `json:"visibilityTimeout,omitempty"`                                                     // Visibility timeout in seconds
+	Arn                string           `json:"arn,omitempty" camel:"QUEUENAMEORARN"`               // SQS ARN
+	DeleteAfterRead    bool             `json:"deleteAfterRead" default:"true"`                     // Auto-delete messages after reading
+	AutoCreateQueue    bool             `json:"autoCreateQueue" default:"false"`                    // Auto-create SQS queue
+	Host               string           `json:"host" camel:"AMAZONAWSHOST" default:"amazonaws.com"` // AWS host
+	Protocol           string           `json:"protocol" default:"https"`                           // Communication protocol (http/https)
+	QueueURL           string           `json:"queueURL,omitempty"`                                 // Full SQS queue URL
+	Greedy             bool             `json:"greedy" default:"false"`                             // Greedy scheduler
+	Delay              int              `json:"delay" default:"500"`                                // Delay between polls in milliseconds
+	MaxMessagesPerPoll int              `json:"maxMessagesPerPoll" default:"1"`                     // Max messages to return (1-10)
+	WaitTimeSeconds    int              `json:"waitTimeSeconds,omitempty"`                          // Wait time for messages
+	VisibilityTimeout  int              `json:"visibilityTimeout,omitempty"`                        // Visibility timeout in seconds
 }
 
 type AWSDDBStreams struct {
@@ -71,6 +71,6 @@ type AWSDDBStreams struct {
 
 type AWSSNS struct {
 	AWSCommon       `json:",inline"` // Embeds AWSCommon to inherit its fields in JSON
-	Arn             string           `json:"arn,omitempty" camel:"CAMEL_KAMELET_AWS_SNS_SINK_TOPICNAMEORARN"` // SNS ARN
-	AutoCreateTopic bool             `json:"autoCreateTopic" default:"false"`                                 // Auto-create SNS topic
+	Arn             string           `json:"arn,omitempty" camel:"TOPICNAMEORARN"` // SNS ARN
+	AutoCreateTopic bool             `json:"autoCreateTopic" default:"false"`      // Auto-create SNS topic
 }

--- a/pkg/reconciler/integration/helper.go
+++ b/pkg/reconciler/integration/helper.go
@@ -54,7 +54,9 @@ func GenerateEnvVarsFromStruct(prefix string, s interface{}) []corev1.EnvVar {
 
 		// First, check for the custom 'camel' tag
 		envVarName := fieldType.Tag.Get("camel")
-		if envVarName == "" {
+		if envVarName != "" {
+			envVarName = fmt.Sprintf("%s_%s", prefix, envVarName)
+		} else {
 			// If 'camel' tag is not present, fall back to the 'json' tag or Go field name
 			jsonTag := fieldType.Tag.Get("json")
 			tagName := strings.Split(jsonTag, ",")[0]

--- a/pkg/reconciler/integration/helper_test.go
+++ b/pkg/reconciler/integration/helper_test.go
@@ -53,25 +53,77 @@ func TestGenerateEnvVarsFromStruct(t *testing.T) {
 
 func TestGenerateEnvVarsFromStruct_S3WithCamelTag(t *testing.T) {
 	type AWSS3 struct {
-		Arn    string `json:"arn,omitempty" camel:"CAMEL_KAMELET_AWS_S3_SOURCE_BUCKETNAMEORARN"`
+		Arn    string `json:"arn,omitempty" camel:"BUCKETNAMEORARN"`
 		Region string `json:"region,omitempty"`
 	}
 
-	prefix := "CAMEL_KAMELET_AWS_S3_SOURCE"
-	input := AWSS3{
-		Arn:    "arn:aws:s3:::example-bucket",
-		Region: "us-west-2",
+	type AWSSQS struct {
+		Arn    string `json:"arn,omitempty" camel:"QUEUENAMEORARN"`
+		Region string `json:"region,omitempty"`
 	}
 
-	// Expected environment variables including SSL settings and camel tag for Arn
-	want := []corev1.EnvVar{
-		{Name: "CAMEL_KAMELET_AWS_S3_SOURCE_BUCKETNAMEORARN", Value: "arn:aws:s3:::example-bucket"},
-		{Name: "CAMEL_KAMELET_AWS_S3_SOURCE_REGION", Value: "us-west-2"},
+	tests := []struct {
+		name   string
+		prefix string
+		input  interface{}
+		want   []corev1.EnvVar
+	}{
+		{
+			name:   "S3 Source with Camel Tag",
+			prefix: "CAMEL_KAMELET_AWS_S3_SOURCE",
+			input: AWSS3{
+				Arn:    "arn:aws:s3:::example-bucket",
+				Region: "us-west-2",
+			},
+			want: []corev1.EnvVar{
+				{Name: "CAMEL_KAMELET_AWS_S3_SOURCE_BUCKETNAMEORARN", Value: "arn:aws:s3:::example-bucket"},
+				{Name: "CAMEL_KAMELET_AWS_S3_SOURCE_REGION", Value: "us-west-2"},
+			},
+		},
+		{
+			name:   "S3 Sink with Camel Tag",
+			prefix: "CAMEL_KAMELET_AWS_S3_SINK",
+			input: AWSS3{
+				Arn:    "arn:aws:s3:::another-bucket",
+				Region: "eu-central-1",
+			},
+			want: []corev1.EnvVar{
+				{Name: "CAMEL_KAMELET_AWS_S3_SINK_BUCKETNAMEORARN", Value: "arn:aws:s3:::another-bucket"},
+				{Name: "CAMEL_KAMELET_AWS_S3_SINK_REGION", Value: "eu-central-1"},
+			},
+		},
+		{
+			name:   "SQS Source with Camel Tag",
+			prefix: "CAMEL_KAMELET_AWS_SQS_SOURCE",
+			input: AWSSQS{
+				Arn:    "arn:aws:sqs:::example-queue",
+				Region: "us-east-1",
+			},
+			want: []corev1.EnvVar{
+				{Name: "CAMEL_KAMELET_AWS_SQS_SOURCE_QUEUENAMEORARN", Value: "arn:aws:sqs:::example-queue"},
+				{Name: "CAMEL_KAMELET_AWS_SQS_SOURCE_REGION", Value: "us-east-1"},
+			},
+		},
+		{
+			name:   "SQS Sink with Camel Tag",
+			prefix: "CAMEL_KAMELET_AWS_SQS_SINK",
+			input: AWSSQS{
+				Arn:    "arn:aws:sqs:::another-queue",
+				Region: "ap-southeast-1",
+			},
+			want: []corev1.EnvVar{
+				{Name: "CAMEL_KAMELET_AWS_SQS_SINK_QUEUENAMEORARN", Value: "arn:aws:sqs:::another-queue"},
+				{Name: "CAMEL_KAMELET_AWS_SQS_SINK_REGION", Value: "ap-southeast-1"},
+			},
+		},
 	}
 
-	got := GenerateEnvVarsFromStruct(prefix, input)
-
-	if diff := cmp.Diff(want, got); diff != "" {
-		t.Errorf("generateEnvVarsFromStruct_S3WithCamelTag() mismatch (-want +got):\n%s", diff)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := GenerateEnvVarsFromStruct(tt.prefix, tt.input)
+			if diff := cmp.Diff(tt.want, got); diff != "" {
+				t.Errorf("GenerateEnvVarsFromStruct() mismatch (-want +got):\n%s", diff)
+			}
+		})
 	}
 }

